### PR TITLE
Fix compiling code using Kokkos::printf for OpenMPTarget on Intel GPUs

### PR DESCRIPTION
**Pull request by masterleinad**
Compiling the unit tests with `OpenMPTargets` on `Intel GPUs` gives warnings like
```
non-string-literal argument used in printf is not supported for spir64
```
and compilation fails. As a workaround make `Kokkos::printf` a no-op in these cases if there are arguments.

### CODE REVIEW

**Review by rgayatri23 — APPROVED** [review]

File: core/src/impl/Kokkos_Printf.hpp [hunk 1]
```diff
@@ -39,11 +39,14 @@ KOKKOS_FUNCTION void printf(const char* format, Args... args) {
   else
     sycl::ext::oneapi::experimental::printf(format, args...);
 #else
-  if constexpr (sizeof...(Args) == 0)
-    ::printf("%s", format);
```
Comment (dalg24) [inline core/src/impl/Kokkos_Printf.hpp:43]: Why was it not just `::printf(format)`?

Comment (masterleinad) [reply core/src/impl/Kokkos_Printf.hpp:43]: See the comment above: 
> Some compilers warn if "args" is empty and format is not a string literal

**Review by ldh4 — APPROVED** [review]

**Review by crtrott — APPROVED** [review]

**Comment by dalg24** [comment]
Did we report this to Intel?

**Comment by dalg24** [comment]
The openmptarget build passed. Ignoring timeouts.

**Pull request merged**